confchange/confchange.go

```diff
@@ -259,8 +259,8 @@ func (c Changer) initProgress(cfg *tracker.Config, trk tracker.ProgressMap, id u
 		// at all (and will thus likely need a snapshot), though the app may
 		// have applied a snapshot out of band before adding the replica (thus
 		// making the first index the better choice).
-		Next:      c.LastIndex,
 		Match:     0,
+		Next:      max(c.LastIndex, 1), // invariant: Match < Next
```

Comment:
Why the `Next` is set to `lastIndex + 1` in reset? Should we make them consistent? 

https://github.com/etcd-io/raft/blob/83d8deca057fef78cf3a86ea70457e4d3255e4da/raft.go#L784

Reply:
I tried `c.LastIndex + 1` here for consistency, but it breaks a few tests. I think the reason is: setting to `lastIndex + 1` precludes sending a `MsgApp` message instantly (the `maybeSendAppend` treats it as "everything is in flight", and won't send any appends until there is a heartbeat reply). So, here we set it to `LastIndex` to guarantee at least one message send. Also, typically this last entry contains the very config change that we're performing here.

Reply:
> setting to lastIndex + 1 precludes sending a MsgApp message instantly

Makes sense.